fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java

```diff
@@ -1589,39 +1591,56 @@ public ShowResultSet process(List<AlterClause> alterClauses, Database db, OlapTa
                         PropertyAnalyzer.PROPERTIES_ENABLE_PERSISTENT_INDEX, false);
                 boolean oldEnablePersistentIndex = olapTable.enablePersistentIndex();
                 if (oldEnablePersistentIndex == enablePersistentIndex) {
                     LOG.info(String.format("table: %s enable_persistent_index is %s, nothing need to do",
                             olapTable.getName(), enablePersistentIndex));
                     return null;
                 }
+
+                long timeoutSecond = PropertyAnalyzer.analyzeTimeout(properties, Config.alter_table_timeout_second);
+                alterMetaJob = new LakeTableAlterMetaJob(GlobalStateMgr.getCurrentState().getNextId(),
+                        db.getId(),
+                        olapTable.getId(), olapTable.getName(), timeoutSecond,
+                        TTabletMetaType.ENABLE_PERSISTENT_INDEX, enablePersistentIndex);
+            } else if (properties.containsKey(PropertyAnalyzer.PROPERTIES_DATACACHE_PARTITION_DURATION)) {
+                PeriodDuration partitionDuration = PropertyAnalyzer.analyzeDataCachePartitionDuration(properties);
+                if (partitionDuration == null) {
+                    throw new DdlException("Null datacache.partition_duration");
+                }
+                PeriodDuration oldPartitionDuration = olapTable.dataCachePartitionDuration();
+                if (partitionDuration.equals(oldPartitionDuration)) {
+                    LOG.info("table: {} datacache.partition_duration is {}, nothing need to do",
+                            olapTable.getName(), partitionDuration);
+                    return null;
+                }
+                olapTable.setDataCachePartitionDuration(partitionDuration);
+                ModifyTablePropertyOperationLog info =
+                        new ModifyTablePropertyOperationLog(db.getId(), olapTable.getId(), propClone);
+                GlobalStateMgr.getCurrentState().getEditLog().logAlterTableProperties(info);
             } else {
-                throw new DdlException("only support alter enable_persistent_index in shared_data mode");
+                throw new DdlException("Only support alter enable_persistent_index and datacache.partition_duration " +
+                        "in shared_data mode");
             }
-
-            long timeoutSecond = PropertyAnalyzer.analyzeTimeout(properties, Config.alter_table_timeout_second);
-            alterMetaJob = new LakeTableAlterMetaJob(GlobalStateMgr.getCurrentState().getNextId(),
-                    db.getId(),
-                    olapTable.getId(), olapTable.getName(), timeoutSecond,
-                    TTabletMetaType.ENABLE_PERSISTENT_INDEX, enablePersistentIndex);
         } else {
             // shouldn't happen
-            throw new DdlException("only support alter enable_persistent_index in shared_data mode");
+            throw new DdlException("alterClause is not a instance of ModifyTablePropertiesClause");
```

Comment:
The hit of error maybe not clear, for user don't understand what is `alterClause` and `ModifyTablePropertiesClause` which is internal implemention